src/client/pages/store-management/store-management-view-model.ts

```diff
@@ -0,0 +1,63 @@
+import { PageViewModel, template, route, components, NavigationService } from "@nivinjoseph/n-app";
+import "./store-management-view.scss"; // importing css for the template
+import { Routes } from "../routes";
+import { StoreService } from "../../../sdk/services/store-service/store-service";
+import { inject } from "@nivinjoseph/n-ject";
+import { given } from "@nivinjoseph/n-defensive";
+import { Invoice } from "../../../sdk/proxies/invoice/invoice";
+import { AllInvoicesViewModel } from "./components/all-invoices/all-invoices-view-model";
+import { TotalSalesViewModel } from "./components/total-sales/total-sales-view-model";
+
+
+@template(require("./store-management-view.html")) // path to the template
+@route(Routes.store)// route that the page is going to be shown at
+@components(AllInvoicesViewModel, TotalSalesViewModel)
+@inject("StoreService","NavigationService") //  dependency for this VM, this gets injected in the constructor, these dependencies are installed in client.ts
+export class StoreManagementViewModel extends PageViewModel
+{
+    private readonly _storeService: StoreService;
+    private readonly _navigationService: NavigationService;
+    
+    
+    private readonly _invoices: Array<Invoice> = [];
+    private _selectedTab: string;
+    private _selectedNav: string;
+    
+    
+    public get invoices(): Array<Invoice> { return this._invoices; }
+    public get selectedTab(): string { return this._selectedTab; }
+    public get selectedNav(): string { return this._selectedNav; }
+    
+    
+    public constructor(storeService: StoreService, navigationService: NavigationService)
+    {
+        super();
+        given(storeService, "storeService").ensureHasValue().ensureIsObject();
+        given(navigationService, "navigationService").ensureHasValue().ensureIsObject();
+        
+        this._storeService = storeService;
+        this._navigationService = navigationService;
+   
+        this._selectedTab = "homeTab";
+        this._selectedNav = "";
+    }
+    
+    
+    public createInvoice(): void
+    {
+        this._storeService.createInvoice();
+        this._navigationService.navigate(Routes.invoicePage);
+    }
+
+    public viewTotalSales(): void
```

Comment:
using common method and pass the tab from view is also a good option. instead of writing different methods. it will help you when there is more number of tabs.

Reply:
done